### PR TITLE
Add a new row type in query

### PR DESCRIPTION
**Pull request by rakshith91**

### CODE REVIEW

File: sdk/monitor/azure-monitor-query/tests/test_logs_client.py [hunk 1]
```diff
@@ -245,3 +245,22 @@ def test_logs_query_result_iterate_over_tables():
...
+        query,
+        timespan=None,
+    )
+
+    ## should iterate over tables
+    for table in response:
+        assert table.__class__ == LogsTable
+
+        row = table.row
+        assert row.__class__ == LogsTableRow
```
Comment (rakshith91) [inline sdk/monitor/azure-monitor-query/tests/test_logs_client.py:266]:
```suggestion
        for row in table.rows:
            assert row.__class__ == LogsTableRow
```

File: sdk/monitor/azure-monitor-query/tests/async/test_logs_client_async.py [hunk 2]
```diff
@@ -202,3 +202,23 @@ async def test_logs_query_result_iterate_over_tables():
...
+        query,
+        timespan=None,
+    )
+
+    ## should iterate over tables
+    for table in response:
+        assert table.__class__ == LogsTable
+
+        row = table.row
+        assert row.__class__ == LogsTableRow
```
Comment (rakshith91) [inline sdk/monitor/azure-monitor-query/tests/async/test_logs_client_async.py:224]:
```suggestion
        for row in table.rows:
            assert row.__class__ == LogsTableRow
```

File: sdk/monitor/azure-monitor-query/azure/monitor/query/_metrics_query_client.py [hunk 3]
```diff
@@ -148,8 +148,8 @@ def list_metric_namespaces(self, resource_uri, **kwargs):
             **kwargs)
 
     @distributed_trace
-    def list_metric_definitions(self, resource_uri, metric_namespace=None, **kwargs):
-        # type: (str, str, Any) -> ItemPaged[MetricDefinition]
+    def list_metric_definitions(self, resource_uri, **kwargs):
```
Comment (xiangyan99) [inline sdk/monitor/azure-monitor-query/azure/monitor/query/_metrics_query_client.py:151]: Breaking change. Do you want to add it into changelog?

File: sdk/monitor/azure-monitor-query/azure/monitor/query/_models.py [hunk 4]
```diff
@@ -48,6 +55,40 @@ def _from_generated(cls, generated):
...
+
+    ivar list row: The collection of values in the row.
+    ivar int row_index: The index of the row in the table
+    """
+    def __init__(self, **kwargs):
+        # type: (Any) -> None
+        _col_types = kwargs.pop('col_types', None)
+        row = kwargs.get('row', None)
+        self.row = process_row(_col_types, row)
+        self.row_index = kwargs.get('row_index', None)
```
Comment (xiangyan99) [inline sdk/monitor/azure-monitor-query/azure/monitor/query/_models.py:69]: Is it by design that row_index is null-able?

Comment (rakshith91) [reply sdk/monitor/azure-monitor-query/azure/monitor/query/_models.py]: none of these can be none - changed all of them

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**